File: ExamSystem_BE/JWT/DTOs/Auth/ResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace JWT.DTOs.Auth
{
    public class ResetPasswordRequest
    {
        [Required]
        public string Token { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        [RegularExpression(
            @"^(?=.*[A-Za-z])(?=.*\d)(?=.*[^A-Za-z\d]).{6,}$",
            ErrorMessage = "Mật khẩu phải có ít nhất 6 ký tự, bao gồm chữ, số và ký tự đặc biệt.")]
        public string NewPassword { get; set; } = string.Empty;
    }
}

File: examsystem_fe/lib/features/auth/models/reset_password_request.dart
/// Map từ ResetPasswordRequest.cs trong BE:
/// Token + NewPassword
class ResetPasswordRequest {
  final String token;
  final String newPassword;

  const ResetPasswordRequest({
    required this.token,
    required this.newPassword,
  });

  Map<String, dynamic> toJson() => {
    'token': token,
    'newPassword': newPassword,
  };
}
